pkg/sql/logictest/logic.go

```diff
@@ -2320,6 +2347,15 @@ func (t *logicTest) execStatement(stmt logicStatement) (bool, error) {
 		}
 	}
 
+	if err == nil && stmt.expectRoundTrips >= 0 {
+		if err == nil && t.roundTripsCount.Count != int(stmt.expectRoundTrips) {
```

Comment:
It will be more resilient here if we check that the found round trips are <= expectedRoundTrips, and change the logictest comment to reflect that.

Reply:
@solongordon and I spoke about this, I wonder if putting an upper bound is the right approach since it could lead to the case where we decrease the number of roundtrips but then end up increasing it again but we still keep it under the upper bound.